### PR TITLE
Feat: Add class utility to link to look like buttons

### PR DESCRIPTION
**Pull request by lmuntaner**
# Motivation

Add a utility class for the anchor tags to look like a primary button.

This is used in the Buy ICP flow of the NNS Dapp. It is also used with an icon. Therefore, I also added the `with-icon` utility class to both the link and button utility classes.

# Changes

* New class utility for anchor tag: `button` + `primary` + `full-width` + `with-icon`.
* Add `with-icon` to the button utility classes.
* LInk page docs.
* Add a card in Utilities linking to the link docs.
* Add e2e screenshot test for the links docs.
* Add docs about `with-icon` in the button docs.

# Screenshots

Check the screenshots in the PR.

### CODE REVIEW

**Comment by lmuntaner** [comment]
@peterpeterparker @dskloetd please review

File: src/routes/(page)/utility-classes/links/+page.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+## Showcase
+
+### Default
+
+<div class="container" data-tid="showcase">
+
+| Styles                                                                            |
+| --------------------------------------------------------------------------------- |
+| <a href="#">Default Link</a>                                                      |
+| <button class="primary">Primary Button</button>                                   |
```
Comment (dskloetd) [inline src/routes/(page)/utility-classes/links/+page.md:19]: Why is there a button here?

**Review by dskloetd — APPROVED** [review]

File: src/routes/(page)/utility-classes/links/+page.md (hunk 1 above)
Comment (lmuntaner) [reply src/routes/(page)/utility-classes/links/+page.md]: I added it for users to be able to check that it's indeed the same style. But true that it's not necessary...

File: src/lib/styles/global/link.scss [hunk 2]
```diff
@@ -13,4 +13,48 @@ a {
   &:hover {
     color: var(--value-color);
   }
+
+  &.button {
```
Comment (mstrasinskis) [inline src/lib/styles/global/link.scss:17]: Just thinking out loud, would it make sense to create a button `mixin` to share some paddings and colors?

Comment (lmuntaner) [reply src/lib/styles/global/link.scss:17]: Yes, I agree. Let me do it.

Comment (mstrasinskis) [reply src/lib/styles/global/link.scss:18]: Thank you!

File: src/lib/styles/global/button.scss [hunk 3]
```diff
@@ -1,12 +1,17 @@
 @use "../mixins/effect";
 @use "../mixins/fonts";
+@use "../mixins/button";
```
Comment (meodai) [inline src/lib/styles/global/button.scss:3]: i'd rather to a class called .button that can be set on a link than a mixin.

File: src/lib/styles/global/link.scss [hunk 4]
```diff
@@ -13,4 +14,35 @@ a {
...
```
Comment (meodai) [inline src/lib/styles/global/link.scss:18]: why should the class .button only work on links? seams overly specific.

File: src/lib/styles/global/button.scss (hunk 3 above)
Comment (lmuntaner) [reply src/lib/styles/global/button.scss:3]: That's a good point.

I believe that supporting all the other button types would be a bit more work than what I was expecting in this PR.

Comment (meodai) [reply src/lib/styles/global/button.scss:3]: But like this you are actively preventing from it being used anywhere else + the code is there twice, once in the button css and one in tha a.button css.

File: src/lib/styles/global/link.scss (hunk 2 above)
Comment (lmuntaner) [reply src/lib/styles/global/link.scss:18]: done!

**Review by mstrasinskis — APPROVED** [review]
Thanks!

**Pull request merged**